arcgis-ios-sdk-samplesTests/Sample Tests/GraphicsOverlayDictionaryRenderer3DViewControllerTests.swift

```diff
@@ -42,18 +42,36 @@ class GraphicsOverlayDictionaryRenderer3DViewControllerTests: XCTestCase {
 }
 
 class MessageParserTests: XCTestCase {
-    let messagesURL = Bundle.main.url(forResource: "Mil2525DMessages", withExtension: "xml")!
-    
     func testParseMessagesAtURL() {
-        let parser = MessageParser()
+        let messagesURL = Bundle.main.url(forResource: "Mil2525DMessages", withExtension: "xml")!
+        
+        guard let messagesData = try? Data(contentsOf: messagesURL) else {
+            XCTFail("Error loading data from URL + \(messagesURL)")
+            return
+        }
+        
+        let messages: [Message]
         do {
-            let messagesData = try Data(contentsOf: messagesURL)
-            let messages = try parser.parseMessages(from: messagesData)
-            XCTAssertEqual(messages.count, 33)
-            XCTAssertEqual(messages.first?.points.count, 36)
-            XCTAssertEqual(messages.first?.attributes.count, 8)
+            let parser = MessageParser()
+            messages = try parser.parseMessages(from: messagesData)
         } catch {
-            fatalError("Error parsing messages: \(error)")
+            XCTFail("Error parsing messages: \(error)")
+            return
         }
+        
+        let expectedMessageCount = 33
+        let actualMessageCount = messages.count
+        XCTAssertEqual(actualMessageCount, expectedMessageCount)
+
+        let firstMessage = messages.first
+        XCTAssertNotNil(firstMessage)
+        
+        let expectedPointCount = 36
+        let actualFirstMessagePointCount = firstMessage!.points.count
```

Comment:
Assuming there are no messages, force unwrapping `firstMessage` would cause a crash.

```suggestion
        let actualFirstMessagePointCount = firstMessage?.points.count
```

Reply:
I introduced a call to `XCTAssertNotNil(firstMessage)` on line 67 to account for the fact that `firstMessage` could be `nil`. These assertions occur after that potential failure.

Reply:
You're absolutely right, however, that the test could still crash post-assertion if the XML file or the parser were to change subsequently. I'll tidy this up.

Reply:
That is all true. However, we're not setting `continueAfterFailure` and the default value is `true`, so this line would be reached, even if the assertion on line `67` failed. Apple introduced `XCTUnwrap` to solve this problem, but it is only available when using Xcode 11. I'll send you a Xcode 10 friendly implementation you can use if you'd like to go that route.